### PR TITLE
feat(category-selector): add category selector component from Relay

### PR DESCRIPTION
**Pull request by megansmith-box**
ISSUE: https://github.com/box/box-ui-elements/issues/2438
- Adding Category Selector as a component
- Use Component in Message Center
- TODO- Use Category Selector Component in Relay

Screenshots
**New Category Selector Component in Message Center**
category-selector-in-message-center
**Category Selector in Storybook**
Screen Shot 2020-12-28 at 9 22 52 AM
Screen Shot 2020-12-28 at 9 10 06 AM
**Old Category Selector -- NOT FROM COMPONENT -- Original to compare**
category-selector-original
**Focus State for Selected (Blue) Button**
**Focus & Hover State for Non-Selected Button**

### CODE REVIEW

**Review by box-internship-take-home — DISMISSED** [review]

**Review by DanDeMicco — APPROVED** [review]

**Review by alexkrolick — APPROVED** [review]

File: .eslintrc.js [hunk 1]
```diff
@@ -31,8 +31,38 @@ module.exports = {
         {
             files: ['*.ts', '*.tsx'],
             rules: {
+                'flowtype/no-types-missing-file-annotation': 'off',
```
Comment (alexkrolick) [inline .eslintrc.js:34]: ✅ this is ok from the AFT side - we will need to take a look at the gaps in the shared config and address them

File: .eslintrc.js [hunk 2]
```diff
@@ -31,8 +31,38 @@ module.exports = {
...
+                            "object": false, // fixme
+                            "{}": false, // fixme
+                        }
+                    }
+                ],
+                // note you must disable the base rule as it can report incorrect errors:
+                'no-use-before-define': 'off',
+                '@typescript-eslint/ban-ts-ignore': 'off', // deprecated
+                '@typescript-eslint/no-use-before-define': ['error'],
+                '@typescript-eslint/ban-ts-comment': [
```
Comment (alexkrolick) [inline .eslintrc.js:52]: this replaces the deprecated ban-ts-ignore option

File: src/components/category-selector/CategorySelector.scss [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        border-radius: 4 * $bdl-grid-unit;
+        outline: none;
+        cursor: pointer;
+
+        &:hover,
+        &:focus {
+            padding: 2 * $bdl-grid-unit 4 * $bdl-grid-unit;
+            font-weight: bold;
+            background-color: $bdl-gray-05;
+            border: 1px solid transparent;
```
Comment (alexkrolick) [inline src/components/category-selector/CategorySelector.scss:42]: Can you remove the values here that are the same as the base? If someone needs to override those values, they would need to add the :focus and :hover states, too.

```suggestion
            background-color: $bdl-gray-05;
```

Comment (alexkrolick) [reply src/components/category-selector/CategorySelector.scss:42]: Nevermind, this is _because_ of the button component doing that in the first place

**Review by alexkrolick — APPROVED** [review]

**Review by sahiga — APPROVED** [review]

**Pull request merged**